pkg/controller/provider/container/ovirt/client.go

```diff
@@ -64,15 +86,43 @@ func (r *Client) connect() (err error) {
 			TLSClientConfig:       &tls.Config{RootCAs: roots},
 		},
 	}
+
+	url, err := liburl.Parse(r.url)
+	if err != nil {
+		return
+	}
+
+	url.Path = "/ovirt-engine/sso/oauth/token"
+	values := url.Query()
+	values.Add("grant_type", "password")
+	values.Add("username", string(r.secret.Data["user"]))
+	values.Add("password", string(r.secret.Data["password"]))
+	values.Add("scope", "ovirt-app-api")
+
 	client.Header = http.Header{
 		"Accept": []string{"application/json"},
-		"Authorization": []string{
-			"Basic",
-			r.auth()},
-		"Version": []string{"4"},
+	}
+
+	response := &ovirtTokenResponse{}
+	url.RawQuery = values.Encode()
+	client.Get(url.String(), response)
+
+	// Set the access token we received
+	client.Header = http.Header{
+		"Accept":        []string{"application/json"},
+		"Authorization": []string{"Bearer " + response.AccessToken},
+		"Version":       []string{"4"},
 	}
 
 	r.client = client
+	r.created = time.Now()
+
+	expiration, err := strconv.ParseInt(response.Expiration, 10, 64)
+	if err != nil {
+		return
+	}
+
+	r.expiration = time.Now().Local().Add(time.Duration(expiration))
```

Comment:
If the user does not enter correct auth credentials it will fail with...
```
{"level":"info","ts":1660222302.1400778,"logger":"provider","msg":"Connection test failed.","reason":"strconv.ParseInt: parsing \"\": invalid syntax"}
```
Because the `response.Expiration` and `response.AccessToken` are empty strings.
The `msg` is correct but it would be better to have some more proper `reason`.

Reply:
added error check above